Shokofin/Utils/GuardedMemoryCache.cs

```diff
@@ -55,11 +57,7 @@ public TItem GetOrCreate<TItem>(object key, Action<TItem> foundAction, Func<ICac
             return value;
         }
 
-        var semaphore = GetSemaphore(key);
-
-        semaphore.Wait();
-
-        try {
+        using (Semaphores.Lock(key)) {
```

Comment:
I'll give credit where credit is due, and this sir, looks a lot cleaner then it was before.

Reply:
Cheers!